Content.Server/Defusable/Systems/DefusableSystem.cs

```diff
@@ -0,0 +1,435 @@
+    {
+        if (!comp.Usable)
+        {
+            _popup.PopupEntity(Loc.GetString("defusable-popup-fried", ("name", uid)), uid);
+            return;
+        }
+
+        var xform = Transform(uid);
+        if (!xform.Anchored)
+            _transform.AnchorEntity(uid, xform);
+
+        SetBolt(comp, true);
+        SetActivated(comp, true);
+
+        _popup.PopupEntity(Loc.GetString("defusable-popup-begun", ("name", uid)), uid);
+        if (TryComp<OnUseTimerTriggerComponent>(uid, out var timerTrigger))
+        {
+            _trigger.HandleTimerTrigger(
+                uid,
+                null,
+                timerTrigger.Delay,
+                timerTrigger.BeepInterval,
+                timerTrigger.InitialBeepDelay,
+                timerTrigger.BeepSound
+            );
+        }
+
+        RaiseLocalEvent(uid, new BombArmedEvent(uid));
+
+        _appearance.SetData(uid, DefusableVisuals.Active, comp.Activated);
+        _adminLogger.Add(LogType.Explosion, LogImpact.High,
+            $"{ToPrettyString(uid):entity} has begun to countdown.");
+
+        if (TryComp<WiresPanelComponent>(uid, out var wiresPanelComponent))
+            _wiresSystem.TogglePanel(uid, wiresPanelComponent, false);
+    }
+
+    public void TryDetonateBomb(EntityUid uid, DefusableComponent comp)
+    {
+        if (!comp.Activated)
+            return;
+
+        _popup.PopupEntity(Loc.GetString("defusable-popup-boom", ("name", uid)), uid, PopupType.LargeCaution);
+
+        RaiseLocalEvent(uid, new BombDetonatedEvent(uid));
+
+        _explosion.TriggerExplosive(uid);
+        QueueDel(uid);
+
+        _appearance.SetData(uid, DefusableVisuals.Active, comp.Activated);
+
+        _adminLogger.Add(LogType.Explosion, LogImpact.High,
+            $"{ToPrettyString(uid):entity} has been detonated.");
+    }
+
+    public void TryDefuseBomb(EntityUid uid, DefusableComponent comp)
+    {
+        if (!comp.Activated)
+            return;
+
+        _popup.PopupEntity(Loc.GetString("defusable-popup-defuse", ("name", uid)), uid);
+        SetActivated(comp, false);
+
+        var xform = Transform(uid);
+
+        if (comp.Disposable)
+        {
+            SetUsable(comp, false);
+            RemComp<ExplodeOnTriggerComponent>(uid);
+            RemComp<OnUseTimerTriggerComponent>(uid);
+        }
+        RemComp<ActiveTimerTriggerComponent>(uid);
+
+        _audio.PlayPvs(comp.DefusalSound, uid);
+
+        RaiseLocalEvent(uid, new BombDefusedEvent(uid));
+
+        comp.ActivatedWireUsed = false;
+        comp.DelayWireUsed = false;
+        comp.ProceedWireCut = false;
+        comp.ProceedWireUsed = false;
+        comp.Bolted = false;
+
+        if (xform.Anchored)
+            _transform.Unanchor(uid, xform);
+
+        _appearance.SetData(uid, DefusableVisuals.Active, comp.Activated);
+        _adminLogger.Add(LogType.Explosion, LogImpact.High,
+            $"{ToPrettyString(uid):entity} has been defused!");
+    }
```

Comment:
these should take in `User` as a param for admin logging properly, otherwise the logging is sorta useless

Reply:
TryDetonateBomb and TryDefuseBomb are (to my knowledge) only run either through the bombs' timer going off (which does not have a user) or through cutting / pulsing the related wire, and all of those methods already log the user cutting said wire so it should be fine.